pkg/sources/gitlab/gitlab.go

```diff
@@ -324,7 +324,7 @@ func (s *Source) scanRepos(ctx context.Context, chunksChan chan *sources.Chunk,
 				}
 				s.SetProgressComplete(i, len(repos), fmt.Sprintf("Repo: %s", repoURL), "")
 
-				path, repo, err := git.CloneRepoUsingToken(s.token, repoURL.String(), s.user)
+				path, repo, err := git.CloneRepoUsingToken(s.token, repoURL.String(), "oauth2")
```

Comment:
Maybe "placeholder" instead? We don't support oauth2 for gitlab, this is kind of confusing.

Reply:
well the "oauth2" string is actually need in order to server as the user here: https://github.com/trufflesecurity/trufflehog/blob/6d82056a9b76247723cb31758301c6335b0a1e89/pkg/sources/git/git.go#L245

That way when we construct the url to clone it has the correct format of:
`git clone https://oauth2:${Personal Access Tokens}@gitlab.com/username/myrepo.git` Without oauth2 as part of the url string the clone command fails because it requires a username and password.

Reply:
Are you saying the specific string "oauth2" is required and a string that is "placeholder" (or similar) won't work? As far as I can tell, there is nothing related to oauth2 happening here so it's confusing.

Reply:
ahh got it, yea I was under the impression it needed to be oauth2 as the string before `:` in the url because that's what i was reading [Stackover answers](https://stackoverflow.com/questions/25409700/using-gitlab-token-to-clone-without-authentication)

but with some additional testing as long as it's a string it works, it just can't be an empty string. Interesting and weird 🤔  

Updated to "placeholder"